File: desarrollo/Backend/PruebaPlayaSenator/PruebaPlayaSenator.Application/Shared/PaginationOptions.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PruebaPlayaSenator.Application.Shared
{
    public class PaginationOptions
    {
        public int PageSize { get; set; }
        public int CurrentPage { get; set; }
        public int TotalItemsCount { get; set; }
        public int CurrentItemsCount { get; set; }
        public bool HasPreviousPage { get; set; }
        public bool HasNextPage { get; set; }
    }
}

File: desarrollo/FrontEnd/src/app/shared/PaginationOptions.ts
export class PaginationOptions {
    pageSize: number;
    currentPage: number;
    totalItemsCount?: number;
    hasPreviousPage?: number;
    hasNextPage?: number;
    currentItemsCount?: number;
  
    constructor(pageSize: number, currentPage: number, totalItemsCount?: number, hasPreviousPage?: number, hasNextPage?: number, currentItemsCount?:number) {
      this.pageSize = pageSize;
      this.currentPage = currentPage;
      this.totalItemsCount = totalItemsCount;
      this.hasPreviousPage = hasPreviousPage;
      this.hasNextPage = hasNextPage;
      this.currentItemsCount = currentItemsCount;
    }
  }